external-import/zerofox/src/open_cti/observables.py

```diff
@@ -0,0 +1,32 @@
+from typing import Type
+
+from stix2 import TLP_AMBER
+from stix2.base import _Observable
+
+main_observable_type = "x_opencti_main_observable_type"
+created_by_ref = "x_opencti_created_by_ref"
+
+
+def _get_observable_type(cls: Type[_Observable]) -> str:
+    _type = {
+        "IPv4-Address": "IPv4-Addr",
+        "IPv6-Address": "IPv6-Addr",
+        "File": "StixFile",
+        "URL": "Url",
+        "AutonomousSystem": "Autonomous-System",
+        "X509Certificate": "X509-Certificate",
+    }.get(cls.__name__)
+    if not _type:
+        raise ValueError(f"Unsupported observable type: {cls.__name__}")
+    return _type
+
+
+def build_observable(created_by: str, cls: Type[_Observable], **kwargs):
+    return cls(
+        **kwargs,
+        custom_properties={
+            main_observable_type: _get_observable_type(cls),
+            created_by_ref: created_by,
+        },
+        object_marking_refs=[TLP_AMBER.id],
```

Comment:
I've reviewed all the changes following my comments and I have to say it's great! Thanks for all your hard work.

I have one last comment before testing it:

- You should add markings to the entities you create. As you've done for Observables, you should do the same for the others.

NB: You can set TLP AMBER for your proprietary data, and TLP CLEAR for the rest (e.g. for the "ZeroFox" Org).

Reply:
Looks great to me! Thanks